app/models/product_import/entry_validator.rb

```diff
@@ -283,7 +287,19 @@ def product_field_errors(entry, existing_product)
     end
 
     def attributes_match?(attribute, existing_product, entry)
-      existing_product.public_send(attribute) == entry.public_send(attribute)
+      existing_product_value = existing_product.public_send(attribute)
+      entry_value = entry.public_send(attribute)
+      existing_product_value == convert_to_trusted_type(entry_value, existing_product_value)
+    end
+
+    def convert_to_trusted_type(untrusted_attribute, trusted_attribute)
+      if trusted_attribute.is_a? Integer
```

Comment:
I just learned this a few weeks ago:
```ruby
case trusted_attributes
  when Integer #...
  when Float #...
  else #...
end
```

Reply:
awesome, I tried something similar but couldnt make it work. thanks, done.